### PR TITLE
LF: Drop CONTRACT_ERROR

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
part of #8020

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/ExceptionTest.scala [hunk 1]
```diff
@@ -628,9 +578,9 @@ class ExceptionTest extends AnyWordSpec with Matchers with TableDrivenPropertyCh
     )
   }
 
-  private def runExpr(pkgs1: PureCompiledPackages)(e: Expr): SResult = {
-    Speedy.Machine.fromPureExpr(pkgs1, e).run()
-  }
+//  private def runExpr(pkgs1: PureCompiledPackages)(e: Expr): SResult = {
+//    Speedy.Machine.fromPureExpr(pkgs1, e).run()
+//  }
```
Comment (cocreature) [inline daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/ExceptionTest.scala:583]: can we delete this?

File: daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/ExceptionTest.scala [hunk 2]
```diff
@@ -65,53 +62,6 @@ class ExceptionTest extends AnyWordSpec with Matchers with TableDrivenPropertyCh
   //   }
   // }
 
-  // TODO https://github.com/digital-asset/daml/issues/8020
-  //   Add builtin errors back into this test (ArithmeticError and ContractError).
-  //   In these cases the message should probably just be "ArithmeticError" and "ContractError".
-
-  "ANY_EXCEPTION_MESSAGE" should {
-
-    // Application of ANY_EXCEPTION_MESSAGE for various cases:
```
Comment (cocreature) [inline daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/ExceptionTest.scala:74]: why did you delete this?

File: compiler/damlc/daml-stdlib-src/DA/Internal/Exception.daml [hunk 3]
```diff
@@ -121,28 +119,6 @@ instance HasFromAnyException ArithmeticError where
 
 --------------------------------------------------------------
 
-instance HasField "message" ContractError Text where
```
Comment (cocreature) [inline compiler/damlc/daml-stdlib-src/DA/Internal/Exception.daml:124]: We still need this in Daml. We just need to change the instances. Maybe just turn them into `error` calls and then @sofiafaro-da or someone else can fix it up later.

File: compiler/damlc/daml-prim-src/DA/Exception/ContractError.daml [hunk 4]
```diff
@@ -1,24 +0,0 @@
--- Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
--- SPDX-License-Identifier: Apache-2.0
-
-{-# LANGUAGE CPP #-}
```
Comment (cocreature) [inline compiler/damlc/daml-prim-src/DA/Exception/ContractError.daml:4]: same here we still have this exception in Daml.

**Review by cocreature — APPROVED** [review]
lgtm, thanks!

File: daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/ExceptionTest.scala (hunk 1 above)
Comment (remyhaemmerle-da) [reply daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/ExceptionTest.scala]: reverted

File: compiler/damlc/daml-prim-src/DA/Exception/ContractError.daml (hunk 4 above)
Comment (remyhaemmerle-da) [reply compiler/damlc/daml-prim-src/DA/Exception/ContractError.daml]: reverted

File: compiler/damlc/daml-stdlib-src/DA/Internal/Exception.daml [hunk 5]
```diff
@@ -119,6 +121,28 @@ instance HasFromAnyException ArithmeticError where
...
+
+instance HasToAnyException ContractError where
+    toAnyException = primitive @"EToAnyException"
+
+instance HasFromAnyException ContractError where
+    fromAnyException e
+        | primitive @"BEAnyExceptionIsContractError" e =
+            Some (ContractError (message e))
+        | otherwise =
+            primitive @"EFromAnyException" e
```
Comment (sofiafaro-da) [inline compiler/damlc/daml-stdlib-src/DA/Internal/Exception.daml]: Since we don't want the primitive any more, this will just become:

```Haskell
fromAnyException = primitive @"EFromAnyException"
```

And that's all that needs to change in this file. Then we can remove the primitive everywhere from the Haskell side as well.

File: compiler/damlc/daml-stdlib-src/DA/Internal/Exception.daml [hunk 6]
```diff
@@ -135,11 +135,8 @@ instance HasToAnyException ContractError where
     toAnyException = primitive @"EToAnyException"
 
 instance HasFromAnyException ContractError where
-    fromAnyException e
-        | primitive @"BEAnyExceptionIsContractError" e =
-            Some (ContractError (message e))
-        | otherwise =
-            primitive @"EFromAnyException" e
+    fromAnyException e =
+      primitive @"EFromAnyException" e
```
Comment (sofiafaro-da) [inline compiler/damlc/daml-stdlib-src/DA/Internal/Exception.daml:139]:
```suggestion
    fromAnyException = primitive @"EFromAnyException"
```

to keep it consistent with the others :-)

**Review by sofiafaro-da — APPROVED** [review]
LGTM, thank you!
